app/src/main/java/fr/free/nrw/commons/explore/depictions/DepictsClient.java

```diff
@@ -0,0 +1,186 @@
+package fr.free.nrw.commons.explore.depictions;
+
+import androidx.annotation.Nullable;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import fr.free.nrw.commons.BuildConfig;
+import fr.free.nrw.commons.Media;
+import fr.free.nrw.commons.depictions.models.Search;
+import fr.free.nrw.commons.media.MediaInterface;
+import fr.free.nrw.commons.upload.depicts.DepictsInterface;
+import fr.free.nrw.commons.upload.structure.depictions.DepictedItem;
+import fr.free.nrw.commons.utils.CommonsDateUtil;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+
+/**
+ * Depicts Client to handle custom calls to Commons Wikibase APIs
+ */
+@Singleton
+public class DepictsClient {
+
+    private final DepictsInterface depictsInterface;
+    private final MediaInterface mediaInterface;
+    private static final String NO_DEPICTED_IMAGE = "No Image for Depiction";
+
+    @Inject
+    public DepictsClient(DepictsInterface depictsInterface, MediaInterface mediaInterface) {
+        this.depictsInterface = depictsInterface;
+        this.mediaInterface = mediaInterface;
+    }
+
+    /**
+     * Search for depictions using the search item
+     * @return list of depicted items
+     */
+    public Observable<DepictedItem> searchForDepictions(String query, int limit, int offset) {
+        return depictsInterface.searchForDepicts(
+                query,
+                String.valueOf(limit),
+                Locale.getDefault().getLanguage(),
+                Locale.getDefault().getLanguage(),
+                String.valueOf(offset)
+        )
+                .flatMap(depictSearchResponse ->Observable.fromIterable(depictSearchResponse.getSearch()))
+                .map(DepictedItem::new);
+    }
+
+    /**
+     * Get URL for image using image name
+     * Ex: title = Guion Bluford
+     * Url = https://upload.wikimedia.org/wikipedia/commons/thumb/0/04/Guion_Bluford.jpg/70px-Guion_Bluford.jpg
+     */
+    private String getThumbnailUrl(String title) {
```

Comment:
I don't think that this is the right way of getting the thumbnail. It might not work for all images.  `imageInfo` API can be used to fetch the thumbnail.

Reply:
I will need more information to implement this, is there an example usage somewhere